XML_to_SON/grammar_parser_test.py

```diff
@@ -0,0 +1,305 @@
+    for facility_name, facility_config in processed_facilities.items():
+        if "facility" in final_json["simulation"] and facility_name in final_json["simulation"]["facility"]["config"]["ChildExactlyOne"]:
+            new_facility_config = {"InputTmpl": '"'+facility_name+'"'}
+            new_facility_config.update(facility_config)
+            final_json["simulation"]["facility"]["config"][facility_name] = new_facility_config
+
+    for region_name, region_config in processed_regions.items():
+        if "region" in final_json["simulation"] and region_name in final_json["simulation"]["region"]["config"]["ChildExactlyOne"]:
+            new_region_config = {"InputTmpl": '"' + region_name + '"'}
+            new_region_config.update(region_config)
+            final_json["simulation"]["region"]["config"][region_name] = new_region_config
+
+    for institution_name, institution_config in processed_institutions.items():
+        if "region" in final_json["simulation"] and institution_name in final_json["simulation"]["region"]["institution"]["config"]["ChildExactlyOne"]:
+            new_institution_config = {"InputTmpl": '"' + institution_name + '"'}
+            new_institution_config.update(institution_config)
+            final_json["simulation"]["region"]["institution"]["config"][institution_name] = new_institution_config
+
+    return final_json
+
+def custom_format(value):
+    val_str = json.dumps(value)
+    val_str = re.sub(r'^"|"$', '', val_str)
+    val_str = val_str.replace('\\"', '"')
+    return val_str
+
+def custom_serialize(obj, key_name="simulation", indent=0):
+    lines = []
+    base_indent = " " * indent
+    child_indent = " " * (indent + 4)
+
+    lines.append(f"{base_indent}{key_name}={{")
+
+    for key, value in obj.items():
+        if isinstance(value, dict):
+            nested = custom_serialize(value, key, indent + 4)
+            lines.append(nested)
+        else:
+            val_str = custom_format(value)
+            lines.append(f"{child_indent}{key}={val_str}")
+
+    lines.append(f"{base_indent}}}")
+
+    return "\n".join(lines)
+
+def custom_serialize_for_template(obj, annotations, key_name, indent=0):
+    lines = []
+    base_indent = " " * indent
+    doc_indent = " " * (indent + 8) 
+    child_indent = " " * (indent + 4)
+
+    annotation_key = f":agents:{key_name}"
+    if annotation_key not in annotations:
+        matching_keys = [key for key in annotations.keys() if key.endswith(key_name)]
+        if matching_keys:
+            annotation_key = matching_keys[0]
+        else:
+            print(f"No annotations  for {key_name}.")
+            return ""
+        
+    annotation = annotations[annotation_key]
+
+    if 'vars' not in annotation or not isinstance(annotation['vars'], dict):
+        print(f"'vars' missing or not a dictionary in annotations for {key_name}.")
+        return ""
+
+    for key, value in obj.items():
```

Comment:
You don't appear to use the `value` so:
```suggestion
    for key in obj.keys():
```